mssql/compiler.py

```diff
@@ -361,7 +361,7 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                     params.extend(h_params)
 
             explain = self.query.explain_info if django.VERSION >= (4, 0) else self.query.explain_query
-            if explain:
+            if explain and hasattr(self.query, 'explain_format'):
```

Comment:
Hey, so from what I understand after Django 4.0 `query.explain_format` and `query.explain_options` are replaced by `query.explain_info.format` and `query.explain_info.options`  respectively.